src/software/jetson_nano/services/network/network.h

```diff
@@ -51,11 +53,34 @@ class NetworkService
      */
     bool shouldSendNewRobotStatus(const TbotsProto::RobotStatus& robot_status) const;
 
+    /**
+     * Tracks the given primitive set for calculating round-trip time if valid
+     *
+     * @param input A potential primitive set to be logged
+     */
+    void logNewPrimitiveSet(const TbotsProto::PrimitiveSet& new_primitive_set);
+
+    /**
+     * Updates the cached primitive sets for Thunderscope to calculate round-trip time
+     *
+     * @param robot_status The robot status to compare to within the cache
+     */
+    void updatePrimitiveSetLog(TbotsProto::RobotStatus& robot_status);
+
+    /**
+     * Getter for the current epoch time in seconds as a double
+     *
+     * @return current epoch time in seconds as a double
+     */
+    double getCurrentEpochTimeInSeconds();
+
     // Constants
     static constexpr float PROTO_LOSS_WARNING_THRESHOLD          = 0.1f;
     static constexpr unsigned int ROBOT_STATUS_BROADCAST_RATE_HZ = 30;
     static constexpr double ROBOT_STATUS_TO_THUNDERLOOP_HZ_RATIO =
         ROBOT_STATUS_BROADCAST_RATE_HZ / (THUNDERLOOP_HZ + 1.0);
+    static constexpr unsigned int PRIMITIVE_DEQUE_MAX_SIZE =
+        static_cast<unsigned int>(1500 / ROBOT_STATUS_BROADCAST_RATE_HZ);
```

Comment:
I had to stare at this a bit before realizing what this code was doing

add a comment here for people who'll encounter this later, something along the lines of: `// increases size of deque when robot status messages are sent less frequently`